lib/qesapdeployment.pm

```diff
@@ -1261,6 +1261,13 @@ sub qesap_cluster_log_cmds {
             Output => 'azure_cloud_init_log.txt',
         };
     }
+
+    if (get_var('SAVE_LIST_OF_PACKAGES')) {
+        push @log_list, {
+            Cmd => 'rpm -qa',
+            Output => 'rpm-qa-after-patch.txt',
```

Comment:
This is not after patch. This is a list composer for `qesap_cluster_logs` and `qesap_cluster_logs` is not called specifically after the `zypper patch`, not only after that.
I suggest to instead create an API in qesapdeployment.pm called `qesap_dump_paclkages` eventually copying part of qesap_cluster_logs. This function should not have `SAVE_LIST_OF_PACKAGES` check in it. This function could support an argument to be used as log name so that the same API can be called two times from some test module, one like `qesap_dump_packages(name => 'before')` and the other `qesap_dump_packages(name => 'after')`.

About `SAVE_LIST_OF_PACKAGES` I'd try to avoid using any setting in this library, even if at the moment we have some of them about `PUBLICCLOUD`. If needed I'd like all of them to have a specific name prefix. 

Then the test

Reply:
About `SAVE_LIST_OF_PACKAGES` I'd try to avoid using any setting in this library, even if at the moment we have some of them about `PUBLICCLOUD`. If needed I'd like all of them to have a specific name prefix.

Reply:
I'm fine if you like to simplify the change in this file to simply add the new log near all the others, without the setting and with more generic file name, like ...
```
       {
            Cmd => 'cat /var/log/cloudregister',
            Output => 'cloudregister.txt',
        },
       {
            Cmd => 'rpm -qa',
            Output => 'rpm-qa.txt',
        }
);
```

Reply:
> This is not after patch. This is a list composer for `qesap_cluster_logs` and `qesap_cluster_logs` is not called specifically after the `zypper patch`, not only after that.
> 

This is called in `tests/sles4sap/publiccloud/qesap_ansible.pm`, and we do `fully-patch-system` here, so I could also use this function to collect `rpm -qa` results.  There are other places call this function, so I shouldn't name the file to `rpm-qa-after-patch.txt`.


> I suggest to instead create an API in `lib/qesapdeployment.pm` or `lib/sles4sap_publiccloud.pm` called `???_dump_packages`:
> 
> * this new function could eventually copy part of `qesap_cluster_logs`, but as it also has to be executed before  `qesap_exec(ansible)` this should not use Ansible. So it should use ssh but in this case it should use public cloud  `$in->ssh_script_run`, but it is not possible to use PC API `$in->ssh_script_run` from within `lib/qesapdeploymet.pm`. So `lib/qesapdeployment` is not the right place to do it: maybe `lib/sles4sap_publiccloud.pm`
> * This function should not have `SAVE_LIST_OF_PACKAGES` in it. Maybe setting could be at test module level.
> * This function could support an argument to be used as log name so that the same API can be called two times from some test module, one like `qesap_dump_packages(name => 'before')` and the other `???_dump_packages(name => 'after')`.

If I change the file name to `rpm-qa.txt`,  the new function is not needed?